mm2src/coins/utxo.rs

```diff
@@ -1618,16 +1623,23 @@ fn kmd_interest(
     }
 
     // interest stop accruing after 1 year before block 1000000
-    if minutes > 365 * 24 * 60 {
-        minutes = 365 * 24 * 60
+    if minutes > MINUTES_PER_YEAR {
+        minutes = MINUTES_PER_YEAR
     };
     // interest stop accruing after 1 month past 1000000 block
     if height >= 1_000_000 && minutes > 31 * 24 * 60 {
         minutes = 31 * 24 * 60;
     }
-    // next 2 lines ported as is from Komodo codebase
+    // Some of these lines are ported as is from Komodo codebase
     minutes -= 59;
-    let accrued = (value / 10_512_000) * minutes;
+    let mut accrued = (value as f64 * AUR_PER_MINUTE) as u64 * minutes;
```

Comment:
I'm just curious why we need to use floating point arithmetic here, convert from integer to float and back, use `drop_mutability!` macros, rather than using something simpler like:
```rust
    let accrued = if height >= N_S7_HARDFORK_HEIGHT {
        (value / 10_512_000) * minutes / 500
    } else {
        (value / 10_512_000) * minutes
    };
```

Reply:
Thanks for your feedback. The code you provided is better than using `drop_mutability!`, I will fix it in a final fixes PR that I will open.

As for using floating point arithmetic, it was to make the code more understandable as suggested here https://github.com/KomodoPlatform/atomicDEX-API/pull/1841#discussion_r1209446933 , I think I will also go with your suggestion to avoid multiple type castings, I can write a comment above the code to make it more understandable. What do you think @caglaryucekaya since the previous change was your suggestion?

Reply:
I agree that conversion from integer to float is not necessary, but I think using constants is better than having magic numbers around the code. So I suggest the following:

```rust
// MINUTES_PER_YEAR = 365 * 24 * 60
const MINUTES_PER_YEAR: u64 = 525_600;
// Minutes required for 100% active user reward before N_S7_HARDFORK_HEIGHT
const MINUTES_PER_AUR: u64 = 20 * MINUTES_PER_YEAR;
    
let accrued = if height >= N_S7_HARDFORK_HEIGHT {
    (value / MINUTES_PER_AUR) * minutes / 500
} else {
    (value / MINUTES_PER_AUR) * minutes
};

```